File: luci-app-honk/luasrc/model/cbi/honk/dns.lua
local honk = require "luci.model.honk_tools"
local m, s

m = Map("honk", translate("DNS Settings"), translate("Configure DNS settings for HONK."))

local dns_file = "/etc/honk/config.d/dns.dae"

s = honk.init_editor(m, "dns")
honk.add_editor(s, dns_file, "dnsconf", translate("DNS Configuration"))

return m

File: luci-app-honk/root/www/luci-static/resources/view/honk/dns.js
'use strict';
'require honk.common as honk';

return honk.createConfigFileView(
	'/etc/honk/config.d/dns.dae',
	_('DNS Settings'),
	_('Configure DNS settings for HONK.'),
	_('DNS Configuration'),
	_('DNS configuration saved and service reloaded.')
);
